deis/fluentd
# fix(boot): Create better way to generate fluentd conf

closes #31

To manually test this PR do the following:
1. Start with a cluster that is running a fresh workflow dev (or at least very recent)
2. Checkout this PR
3. run `DEIS_REGISTRY=quay.io/ IMAGE_PREFIX=<ACCOUNT> make build push upgrade`
4. Check fluentd logs to make sure fluentd started correctly. You should see something that looks like the following - 

```
<ROOT>
  <source>
    @type tail
    path "/var/log/containers/*.log"
    pos_file "/var/log/containers.log.pos"
    time_format "%Y-%m-%dT%H:%M:%S.%NZ"
    tag "kubernetes.*"
    format json
    read_from_head false
  </source>
  <source>
    type tail
    format syslog
    path "/var/log/startupscript.log"
    pos_file "/var/log/startupscript.log.pos"
    tag "startupscript"
  </source>
  <source>
    type tail
    format /^time="(?<time>[^)]*)" level=(?<severity>[^ ]*) msg="(?<message>[^"]*)"( err="(?<error>[^"]*)")?( statusCode=($<status_code>\d+))?/
    time_format "%Y-%m-%dT%H:%M:%S.%NZ"
    path "/var/log/docker.log"
    pos_file "/var/log/docker.log.pos"
    tag "docker"
  </source>
  <source>
    type tail
    format none
    path "/var/log/etcd.log"
    pos_file "/var/log/etcd.log.pos"
    tag "etcd"
  </source>
  <source>
    type tail
    format multiline
    format_firstline "/^\\w\\d{4}/"
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format "%m%d %H:%M:%S.%N"
    path "/var/log/kubelet.log"
    pos_file "/var/log/kubelet.log.pos"
    tag "kubelet"
  </source>
  <source>
    type tail
    format multiline
    format_firstline "/^\\w\\d{4}/"
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format "%m%d %H:%M:%S.%N"
    path "/var/log/kube-apiserver.log"
    pos_file "/var/log/kube-apiserver.log.pos"
    tag "kube-apiserver"
  </source>
  <source>
    type tail
    format multiline
    format_firstline "/^\\w\\d{4}/"
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format "%m%d %H:%M:%S.%N"
    path "/var/log/kube-controller-manager.log"
    pos_file "/var/log/kube-controller-manager.log.pos"
    tag "kube-controller-manager"
  </source>
  <source>
    type tail
    format multiline
    format_firstline "/^\\w\\d{4}/"
    format1 /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<message>.*)/
    time_format "%m%d %H:%M:%S.%N"
    path "/var/log/kube-scheduler.log"
    pos_file "/var/log/kube-scheduler.log.pos"
    tag "kube-scheduler"
  </source>
  <filter kubernetes.**>
    @type kubernetes_metadata
    kubernetes_url "https://10.167.240.1:443"
    bearer_token_file "/var/run/secrets/kubernetes.io/serviceaccount/token"
    verify_ssl false
  </filter>
  <match **>
    @type copy
    <store>
      @type remote_syslog
      host "10.167.241.98"
      port 514
      log_level warn
    </store>
    <store>
      @type deis
    </store>
  </match>
</ROOT>
```
1. Test that logs come through the system by creating and deploying an application and running `deis logs`.
